### ISSUE TITLE
Return empty object on 204 No Content...

### ISSUE THREAD

**Issue by dandv** [bitinn/node-fetch#165]
...instead of throwing a syntax error due to attempting to JSON parse `''`.

```
let fetch = require('node-fetch');
fetch('http://httpbin.org/status/204').then(res => res.json()).then(object => {
    console.log(object);
}).catch(error => console.error(error));
```

Demo: http://runkit.com/dandv/node-fetch-204-no-content

Fixing this would attain compatibility with [request](https://github.com/request/request/issues/2177) and [got](https://github.com/sindresorhus/got/pull/78).

More generally, it would be more tolerant to return an empty object if the response is the empty string (regardless of the status code being 204 or other 2xx).

**Comment by bedeoverend** [comment]
While I like the idea of an empty object as a JSON response, this moves away from the spec, as they've [consciously stayed with throwing an error](https://github.com/whatwg/fetch/issues/113) - IMO it'd be better to stay with the spec for compatibility with native fetch implementations, but if not, should this at least be added as a known difference?

**Comment by TimothyGu** [comment]
I agree. Plus, there's also the question whether we should return an empty object, or an empty string, or `0`, or `false`: all of which are legitimate JSON types. @bitinn, thoughts?

**Comment by bedeoverend** [comment]
For me, the best thing is to stick with spec - an example use case I'm dealing with is running tests in JSDOM, but now the interface is different between a native client e.g. Chrome and the server side usage. The same applies to `isomorphic-fetch`, if it relies on this library, users will account for different responses based on if its a server or client environment, which essentially breaks the concept of isomorphic.

**Comment by bitinn** [comment]
I am asking browser vendors for opinions in the same WHATWG thread. But status quo seems clear at this point, neither native implementation/browser fetch polyfill/spec handle 204 empty response.

We should revert our code to match their behaviour in `2.x` releases.

**Comment by TimothyGu** [comment]
Agreed. I'll prepare a PR.

**Comment by bedeoverend** [comment]
Thanks very much for the quick response - really appreciate your efforts on this, it's a great polyfill. Cheers!

**Comment by TimothyGu** [comment]
Fixed in #201.
